src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -424,7 +424,7 @@ void ShenandoahConcurrentGC::entry_thread_roots() {
 
 void ShenandoahConcurrentGC::entry_weak_refs() {
   ShenandoahHeap* const heap = ShenandoahHeap::heap();
-  static const char* msg = "Concurrent weak references";
+  const char* msg = conc_weak_refs_event_message();
   ShenandoahConcurrentPhase gc_phase(msg, ShenandoahPhaseTimings::conc_weak_refs);
```

Comment:
Looks good.  With the "static" qualifier, the initialization statement only executes the first time through the function.